From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 21 Jul 2014 11:24:55 +0200
Subject: [224/249] staging: rtl8723au: ODM_DMWatchdog23a(): Take a struct
 hal_data_8723a *
Origin: https://git.kernel.org/linus/ec8884fb20912e866d5f4ce34cdb509c146dbde1

This will allow us to access variables in the hal_data struct
directly, and get rid of the horrendous ODM_CMNINFO spaghetti API.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/hal/odm.c         | 4 +++-
 drivers/staging/rtl8723au/hal/rtl8723a_dm.c | 2 +-
 drivers/staging/rtl8723au/include/odm.h     | 3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/rtl8723au/hal/odm.c b/drivers/staging/rtl8723au/hal/odm.c
index c239b52..e4a9225 100644
--- a/drivers/staging/rtl8723au/hal/odm.c
+++ b/drivers/staging/rtl8723au/hal/odm.c
@@ -287,8 +287,10 @@ void ODM23a_DMInit(struct dm_odm_t *pDM_Odm)
 /*  2011/09/20 MH This is the entry pointer for all team to execute HW out source DM. */
 /*  You can not add any dummy function here, be care, you can only use DM structure */
 /*  to perform any new ODM_DM. */
-void ODM_DMWatchdog23a(struct dm_odm_t *pDM_Odm)
+void ODM_DMWatchdog23a(struct hal_data_8723a *pHalData)
 {
+	struct dm_odm_t *pDM_Odm = &pHalData->odmpriv;
+
 	/* 2012.05.03 Luke: For all IC series */
 	odm_CmnInfoHook_Debug23a(pDM_Odm);
 	odm_CmnInfoUpdate_Debug23a(pDM_Odm);
diff --git a/drivers/staging/rtl8723au/hal/rtl8723a_dm.c b/drivers/staging/rtl8723au/hal/rtl8723a_dm.c
index 204b4ad..272c1e7 100644
--- a/drivers/staging/rtl8723au/hal/rtl8723a_dm.c
+++ b/drivers/staging/rtl8723au/hal/rtl8723a_dm.c
@@ -216,7 +216,7 @@ rtl8723a_HalDmWatchDog(
 		bLinked = true;
 
 	ODM_CmnInfoUpdate23a(&pHalData->odmpriv, ODM_CMNINFO_LINK, bLinked);
-	ODM_DMWatchdog23a(&pHalData->odmpriv);
+	ODM_DMWatchdog23a(pHalData);
 
 skip_dm:
 
diff --git a/drivers/staging/rtl8723au/include/odm.h b/drivers/staging/rtl8723au/include/odm.h
index 546cbb5..b2d8983 100644
--- a/drivers/staging/rtl8723au/include/odm.h
+++ b/drivers/staging/rtl8723au/include/odm.h
@@ -999,7 +999,8 @@ u32 ODM_Get_Rate_Bitmap23a(struct dm_odm_t *pDM_Odm, u32 macid, u32 ra_mask, u8
 
 void ODM23a_DMInit(struct dm_odm_t *pDM_Odm);
 
-void ODM_DMWatchdog23a(struct dm_odm_t *pDM_Odm); /*  For common use in the future */
+struct hal_data_8723a;
+void ODM_DMWatchdog23a(struct hal_data_8723a *pHalData);
 
 void ODM_CmnInfoInit23a(struct dm_odm_t *pDM_Odm, enum odm_cmninfo	CmnInfo, u32 Value);
 
